### ISSUE TITLE
Plugin breaks (some) attachment urls by encoding query string

### ISSUE THREAD

**Issue by GiacoCorsiglia** [deliciousbrains/wp-amazon-s3-and-cloudfront#344]
Hello!

My setup uses a CDN which takes image sizing query parameters, so I filter image urls to include those parameters.  Unfortunately the latest version of Offload S3 breaks those urls by encoding them.  For example:
`.../header.jpg?w=2400&fit=max`
becomes
`.../header1.jpg%3Fw%3D2400%26fit%3Dmax`
which 404s, since all of the encoded stuff is now considered part of the filename by the CDN server.

I think the fix will be very simple.  Change 
`$file_name = basename( $file );`
to
`$file_name = basename( $url['path'] );`
in `Amazon_S3_And_CloudFront::encode_filename_in_path()`.

Thanks!

**Comment by A5hleyRich** [comment]
I'm unable to replicate this. Can you confirm where you're filtering the image URLs?

**Comment by GiacoCorsiglia** [comment]
Thank you for taking a look!

I'm using the [image_downsize](https://developer.wordpress.org/reference/hooks/image_downsize/) filter.

Here's a really short version that should have the same results.  Obviously not correct, but the way I build the url is the same.
```php
add_filter( 'image_downsize', 1, 3, function( $downsize, $attachment_id, $size = 'large' ) {
  $size = is_array( $size ) ? $size : [ 500, 500 ];
  return [
    wp_get_attachment_url( $attachment_id ) . "?w={$size[0]}&h={$size[1]}",
    $size[0],
    $size[1],
    true,
  ];
} );
```

**Comment by A5hleyRich** [comment]
Awesome, I'm able to replicate now. We're looking into a fix.

**Comment by A5hleyRich** [comment]
Fixed in 1.1.1

**Comment by GiacoCorsiglia** [comment]
Was deleting the workaround I had implemented and remembered I forgot to say thanks!  Much appreciated fix
